zerver/lib/outgoing_webhook.py

```diff
@@ -119,23 +122,32 @@ def do_rest_call(rest_operation, request_data, event, service_handler, timeout=N
     request_kwargs['timeout'] = timeout
 
     try:
-        response = requests.request(http_method, final_url, data=json.dumps(request_data), **request_kwargs)
+        response = requests.request(http_method, final_url, data=request_data, **request_kwargs)
         if str(response.status_code).startswith('2'):
-            response_data = service_handler.process_success(response, event)
-            succeed_with_message(event, response_data["response_message"])
+            (response_required, response_message) = service_handler.process_success(response, event)
+            if response_required:
+                succeed_with_message(event, response_message)
 
         # On 50x errors, try retry
         elif str(response.status_code).startswith('5'):
-            request_retry(event, "unable to connect with the third party.")
+            request_retry(event, "Internal Server error at third party.")
         else:
-            response_data = service_handler.process_failure(response, event)
-            fail_with_message(event, response_data["response_message"])
+            (response_required, response_message) = service_handler.process_failure(response, event)
+            if response_required:
+                fail_with_message(event, response_message)
 
     except requests.exceptions.Timeout:
         logging.info("Trigger event %s on %s timed out. Retrying" % (event["command"], event['service_name']))
-        request_retry(event, 'unable to connect with the third party.')
+        request_retry(event, 'Unable to connect with the third party.')
 
     except requests.exceptions.RequestException as e:
         response_message = "An exception occured for message `%s`! See the logs for more information." % (event["command"],)
         logging.exception("Outhook trigger failed:\n %s" % (e,))
         fail_with_message(event, response_message)
+
+def remove_mention_from_message_content(message_content):
+    # type: (Text) -> Text
+    TRIGGER_RE = re.compile(u'(@\*\*.*\*\* )(.*)')
```

Comment:
- This has the effect of stripping off ALL leading mentions; is this intended?
- If the at mention is in the middle of the message, this will drop everything before the mention; is this intended?  (For example, `hello @**Zulip Bot** how are you?` will become `how are you`)
- Should the trailing space be replaced by `\s+`?
- Are there other mention formats besides `@**mention**` that should be supported?  (The answer might be no)